server/blob/blob-union/src/test/java/org/apache/james/blob/union/UnionBlobStoreTest.java

```diff
@@ -489,4 +489,17 @@ void streamHasContentShouldKeepStreamEmptyWhenStreamIsEmpty() throws Exception {
         assertThat(pushBackIS)
             .hasSameContentAs(new ByteArrayInputStream(new byte[0]));
     }
+
+    @Test
+    void deleteBucketShouldDeleteBothCurrentAndLegacyBuckets() {
```

Comment:
In theory, you need 4 tests for that method:
* None exists
* current exists but legacy does not
* legacy exists but current does not
* both exist